nodejs/week2/app.js

```diff
@@ -0,0 +1,80 @@
+// @ts-ignore
+const express = require('express')
+const router = express.Router
+const app = express()
+const port = process.env.PORT || 3000
+const document = require("./documents.json")
+
+
+// Support parsing JSON requests
+app.use(express.json())
+
+// 1:GET /search  ex: GET /search?q=hello:
+app.get('/search',  (req, res) => {
+    try{
+    // @ts-ignore
+    const result =  document.filter(element => Object.values(element).find(word => String(word).includes(req.query.q)));
+   if(result){
+    res.send(result)
+}else{
+    res.sendStatus(404).json({message: "Record not found"});
+}
+}catch(error){
+    res.send(error)
+}
+})
+  
+// 2:GET /documents/:id
+// @ts-ignore
+app.get('/documents/:id', (req, res) => {
+    try{
+    var result =  document.filter(it => it.id === Number(req.params.id));
+    if(result){
+    res.send(result);
+}else{
+    res.sendStatus(404).json({message: "Record not found"});
+}
+}catch(error){
+throw error;
+}
+}),
+ 
+// 3:POST /search
+app.post('/search',  (req, res) => {
+    
+    const fields = req.body.fields;
+    var searchResult ;
+
+    if(fields){
+    // @ts-ignore
```

Comment:
Not a good idea to use //ts-ignore but instead we should fix typescript error if there are any.